webapp/web-components/cht-form/src/app.component.ts

```diff
@@ -0,0 +1,183 @@
+    }
+    this._formXml = value;
+    this.queueRenderForm();
+  }
+
+  @Input() set contactSummary(value: Record<string, any> | undefined) {
+    this._contactSummary = value ? { context: value } : undefined;
+    this.queueRenderForm();
+  }
+
+  @Input() set content(value: Record<string, any> | null) {
+    this._content = value;
+    if (this._content?.contact && !this._content.source) {
+      this._content.source = 'contact';
+    }
+    this.queueRenderForm();
+  }
+
+  @Input() set user(user: Record<string, any>) {
+    if (!user) {
+      throw new Error('The user must be populated.');
+    }
+    this._user = { ...this.DEFAULT_USER, ...user };
+    this.queueRenderForm();
+  }
+
+  get formId(): string {
+    return this._formId;
+  }
+
+  cancelForm(): void {
+    this.tearDownForm();
+    this.onCancel.emit();
+  }
+
+  async submitForm(): Promise<void> {
+    this.status.saving = true;
+
+    try {
+      const currentForm = this.enketoService.getCurrentForm();
+      const formDoc = {
+        xml: this._formXml,
+        doc: {}
+      };
+      const contact = null;  // Only used for setting `from` and `contact` fields on docs
+      const submittedDocs = await this.enketoService.completeNewReport(this._formId, currentForm, formDoc, contact);
+      this.tearDownForm();
+      this.onSubmit.emit(submittedDocs);
+    } catch (e) {
+      console.error('Error submitting form data: ', e);
+      this.status.error = await this.translateService.get('error.report.save');
+    } finally {
+      this.status.saving = false;
+    }
+  }
+
+  private queueRenderForm() {
+    // Ensure that only one render is happening at a time
+    this.currentRender = this.currentRender.then(() => this.renderForm());
+  }
+
+  private async renderForm() {
+    this.unloadForm();
+    if (!this._formHtml || !this._formModel || !this._formXml) {
+      return;
+    }
+
+    const editedListener = () => this.editing = true;
+    const valuechangeListener = () => this.status.error = null;
+    const selector = `#${this._formId}`;
+    const formDoc = { _id: this._formId };
+    const formContext = new EnketoFormContext(selector, 'report', formDoc, this._content);
+    formContext.editedListener = editedListener;
+    formContext.valuechangeListener = valuechangeListener;
+    formContext.contactSummary = this._contactSummary;
+    const formDetails = this.getFormDetails();
+
+    await this.enketoService.renderForm(formContext, formDetails, this._user);
+  }
+
+  private unloadForm() {
+    const currentForm = this.enketoService.getCurrentForm();
+    if (currentForm) {
+      this.enketoService.unload(currentForm);
+    }
+  }
+
+  private getFormDetails() {
+    const $html = $(this._formHtml!);
+    const hasContactSummary = $(this._formModel!)
+      .find('> instance[id="contact-summary"]')
+      .length === 1;
+    return {
+      html: $html,
+      model: this._formModel,
+      // title: form.title,
```

Comment:
Is this property still needed?
It doesn't seem to be used by the enketo service but there is an optional `titleKey` property that can be passed if needed

Reply:
I have removed this line.  I guess the enketo service would use the `title` value from this object as a way to [set a localized form title](https://github.com/medic/cht-core/blob/771850e02c3e7c878931ed9bb08a94768be09ade/webapp/src/ts/services/enketo.service.ts#L212) based on the title value form the actual `form:..`doc in  the db.  But it is not required (and there does not seem to be a need to support this functionality in the web component (at least not in the MVP)).